jerry-core/parser/js/js-parser.c

```diff
@@ -2533,6 +2533,12 @@ parser_parse_class_fields (parser_context_t *context_p) /**< context */
     if (class_field_type & PARSER_CLASS_FIELD_NORMAL)
     {
       parser_emit_cbc_literal (context_p, CBC_ASSIGN_PROP_THIS_LITERAL, literal_index);
+
+      /* Prepare stack slot for assignment property reference base */
+      if (context_p->stack_limit == context_p->stack_depth)
+      {
+        context_p->stack_limit++;
```

Comment:
isn't something like
```c
        if (context_p->stack_limit > PARSER_MAXIMUM_STACK_LIMIT)
        {
          parser_raise_error (context_p, PARSER_ERR_STACK_LIMIT_REACHED);
        }
```
needed here?

Reply:
Maybe we need the assert after all :D
stack_depth can be 1 or 2, so reaching PARSER_MAXIMUM_STACK_LIMIT is impossible

Reply:
Indeed, I think the assert would be enough.